src/main/java/frc/robot/subsystems/swerve/module/ModuleUtils.java

```diff
@@ -0,0 +1,83 @@
+package frc.robot.subsystems.swerve.module;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import frc.utils.Conversions;
+
+public class ModuleUtils {
+
+	public enum ModulePosition {
+
+		FRONT_LEFT(0),
+		FRONT_RIGHT(1),
+		BACK_LEFT(2),
+		BACK_RIGHT(3);
+
+		private final int index;
+
+		ModulePosition(int index) {
+			this.index = index;
+		}
+
+		public int getIndex() {
+			return index;
+		}
+
+	}
+
+	public static double velocityToOpenLoopVoltage(
+		double velocityMetersPerSecond,
+		Rotation2d steerVelocityPerSecond,
+		double couplingRatio,
+		Rotation2d maxVelocityPerSecond,
+		double wheelDiameterMeters,
+		double voltageCompensationSaturation
+	) {
+		Rotation2d velocityPerSecond = Conversions.distanceToAngle(velocityMetersPerSecond, wheelDiameterMeters);
+		Rotation2d coupledVelocityPerSecond = getCoupledAngle(velocityPerSecond, steerVelocityPerSecond, couplingRatio);
+		return velocityToVoltage(coupledVelocityPerSecond, maxVelocityPerSecond, voltageCompensationSaturation);
+	}
+
+	public static double velocityToVoltage(Rotation2d velocityPerSecond, Rotation2d maxVelocityPerSecond, double voltageCompensationSaturation) {
+		double power = velocityPerSecond.getRotations() / maxVelocityPerSecond.getRotations();
+		return Conversions.compensatedPowerToVoltage(power, voltageCompensationSaturation);
+	}
+
+
+	/**
+	 * When the steer motor moves, the drive motor moves as well due to the coupling. This will affect the current position of the drive motor,
+	 * so we need to remove the coupling from the velocity or the position.
+	 *
+	 * @param driveCoupledAngle the position or velocity
+	 * @param steerAngle        the angle or velocity in angle of the module
+	 * @return the distance or velocity without the coupling
+	 */
+	public static Rotation2d getUncoupledAngle(Rotation2d driveCoupledAngle, Rotation2d steerAngle, double couplingRatio) {
```

Comment:
if it could be a velocity or a position you need a name that represents both, angle doesn't represent velocity

Reply:
The velocity is in angle units, position is in angle, velocity in angle per time